### ISSUE TITLE
Possible bug: no-return-await flagging pseudo-returns

### ISSUE THREAD

**Issue by helixbass** [helixbass/eslint-plugin-coffee#16]
I don't really understand `no-return-assign` to start with, but this seems like it's incorrectly flagging these awaits:
```
x = switch (4)
  when 2
    await winning(1)
  when 4
    await winning(5)
```

Which maybe indicates the need to distinguish between "compiled-closure" "returns" vs actual implicit return "returns"?
